homeassistant/components/recorder/util.py

```diff
@@ -373,16 +374,54 @@ def _fail_unsupported_version(
     raise UnsupportedDialect
 
 
+@callback
+def _async_delete_issue_deprecated_version(
+    hass: HomeAssistant, dialect_name: str
+) -> None:
+    """Delete the issue about upcoming unsupported database version."""
+    ir.async_delete_issue(hass, DOMAIN, f"{dialect_name}_too_old")
+
+
+@callback
+def _async_create_issue_deprecated_version(
+    hass: HomeAssistant,
+    server_version: AwesomeVersion,
+    dialect_name: str,
+    min_version: AwesomeVersion,
+) -> None:
+    """Warn about upcoming unsupported database version."""
+    ir.async_create_issue(
+        hass,
+        DOMAIN,
+        f"{dialect_name}_too_old",
+        is_fixable=False,
+        severity=ir.IssueSeverity.CRITICAL,
+        translation_key=f"{dialect_name}_too_old",
+        translation_placeholders={
+            "server_version": str(server_version),
+            "min_version": str(min_version),
+        },
+        breaks_in_ha_version="2025.2.0",
```

Comment:
It's a bit weird to hard code this here when everything else is an input parameter.

Reply:
Its only used in this function.  It didn't seem like we would ever use it any other place.  I can make it a constant if you prefer though.

Reply:
It's fine, it's just a very weird mix of hard coding and input parameters. I guess the `_async_create_issue_deprecated_version` function is there instead of just hard coding everything for when we want to bump the min version also for other engines? But no matter, we can sort that when the need is there.